src/resolve/resolved-dns-packet.c

```diff
@@ -47,13 +50,14 @@ int dns_packet_new(DnsPacket **ret, DnsProtocol protocol, size_t mtu) {
 
         assert(ret);
 
-        if (mtu <= UDP_PACKET_HEADER_SIZE)
+        /* When dns_packet_new() is called with mtu == 0, allocate more than the
+         * absolute minimum (which is the dns packet header size), to avoid
+         * resizing immediately again after appending the first data to the packet.
+         */
+        if (mtu < UDP_PACKET_HEADER_SIZE)
```

Comment:
Instead of doing these change, we should just have partially revert https://github.com/keszybz/systemd/commit/a0166609f782da91710dea9183d1bf138538db37 and also rename the parameter named mtu of dns_packet_new()